qualtran/bloqs/factoring/ecc/ec_add_r.py

```diff
@@ -67,6 +85,41 @@ def signature(self) -> 'Signature':
             [Register('ctrl', QBit()), Register('x', QUInt(self.n)), Register('y', QUInt(self.n))]
         )
 
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', ctrl: Soquet, x: Soquet, y: Soquet
+    ) -> Dict[str, 'SoquetT']:
+        if isinstance(self.n, sympy.Expr):
+            raise DecomposeTypeError("Cannot decompose symbolic `n`.")
+
+        a = bb.add(IntState(bitsize=self.n, val=0))
+        b = bb.add(IntState(bitsize=self.n, val=0))
+
+        ctrl, a = bb.add(XorK(QUInt(bitsize=self.n), self.R.x).controlled(), ctrl=ctrl, x=a)
+        ctrl, b = bb.add(XorK(QUInt(bitsize=self.n), self.R.y).controlled(), ctrl=ctrl, x=b)
+
+        lam_num = (3 * self.R.x**2 + self.R.curve_a) % self.R.mod
+        lam_denom = (2 * self.R.y) % self.R.mod
+
+        lam = (lam_num * pow(lam_denom, -1, mod=self.R.mod)) % self.R.mod
+        lam_r = bb.add(IntState(bitsize=self.n, val=lam))
+
+        a, b, x, y, lam_r = bb.add(
+            ECAdd(self.n, self.R.mod, self.R.curve_a), a=a, b=b, x=x, y=y, lam=lam_r
+        )
+
+        # Call graph doesn't like the controlled() operator on arbitrary clifford.
```

Comment:
what does this mean? does this decomposition not work?

Reply:
Fixed bug in decomposition that was triggering this I think it's okay now.